mt-batcher/services/sequencer/driver.go

```diff
@@ -234,6 +236,22 @@ func (d *Driver) GetBatchBlockRange(ctx context.Context) (*big.Int, *big.Int, er
 	return start, end, nil
 }
 
+func (d *Driver) GetRollupTimeInterval(ctx context.Context, start *big.Int) (time.Duration, error) {
```

Comment:
```
func (d *Driver) RollupTimeInterval(ctx context.Context) (*big.Int, *big.Int, error) {
	log.Debug("RollupTimeInterval start")
	pollingInterval := 500 * time.Millisecond
	rollupTimout := time.Minute * 15
	exit := time.NewTimer(rollupTimout)
	ticker := time.NewTicker(pollingInterval)
	for {
		// normal logic
		start, end, err := d.GetBatchBlockRange(d.Ctx)
		if err != nil {
			return nil, nil, err
		}
		return start, end, nil

		select {
		case <-exit.C:
			// timeout exit
		case <-ticker.C:
		}
	}
}
```
You can use this to do it.

Reply:
fixed.  And add function `GetBatchBlockRangeWithTimeout`